infra/azure/main.tf

```diff
@@ -418,13 +418,24 @@ resource "azuread_application" "auth" {
       type = "Role"
     }
   }
+
+  web {
+    redirect_uris = ["https://auth.${var.domain}:5000/oauth2callback", "http://127.0.0.1:5000/oauth2callback"]
+
+    implicit_grant {
+      access_token_issuance_enabled = true
+      id_token_issuance_enabled     = true
+    }
+  }
 }
 module "auth_sp" {
   source = "./service_principal"
   application_id = azuread_application.auth.application_id
   object_id      = azuread_application.auth.object_id
 }
 
+# FIXME: Create secret /auth-oauth2-client-secret/client_secret.json which is just a copy of the auth service principal credentials
```

Comment:
So, k8s stuff goes in ../k8s/main.tf. There's several examples of `resource kuberenets_secret` there. Pursuant to my other comment, this should be a fresh app registration (not auth's) and not a user's service principal. As such, it should not use the `for_each` thing that `gsa_key` does.